src/scroll/systems/bind.rs

```diff
@@ -0,0 +1,107 @@
+            match scroll_bar.orientation {
+                ScrollBarOrientation::Vertical => {
+                    if scroll_threshold.y != 0.0 {
+                        scroll_handle.visible = true;
+
+                        let handle_length_ratio = root_size.y / content_size.y;
+                        let handle_length = handle_length_ratio * scroll_bar_size.y;
+
+                        scroll_handle.length = handle_length;
+                    } else {
+                        scroll_handle.visible = false;
+                    }
+                }
+                ScrollBarOrientation::Horizontal => {
+                    if scroll_threshold.x != 0.0 {
+                        scroll_handle.visible = true;
+
+                        let handle_length_ratio = root_size.x / content_size.x;
+                        let handle_length = handle_length_ratio * scroll_bar_size.x;
+
+                        scroll_handle.length = handle_length;
+                    } else {
+                        scroll_handle.visible = false;
+                    }
+                }
+            }
+
+            let ratio = scroll_bar_size / scroll_content_node.size();
+            match scroll_bar.orientation {
+                ScrollBarOrientation::Vertical => {
+                    scroll_handle.position = scroll_content.current_offset.y * ratio.y;
+                }
+                ScrollBarOrientation::Horizontal => {
+                    scroll_handle.position = scroll_content.current_offset.x * ratio.x;
+                }
+            }
+        }
+    }
+}
+
+pub fn scroll_handles_that_control_scroll_content(
+    changed_scroll_handles: Query<
+        (&ScrollHandle, &ControlsScrollContent, &Parent),
+        Or<(Changed<ScrollHandle>, Changed<Node>)>,
+    >,
+    scroll_bars_query: Query<(&ScrollBar, &Node)>,
+    mut scroll_content_query: Query<(&mut ScrollContent, &Node)>,
+) {
+    for (scroll_handle, controller, scroll_handle_parent) in changed_scroll_handles.iter() {
+        let (mut scroll_content, scroll_content_node) = scroll_content_query
+            .get_mut(controller.scroll_content)
+            .expect("ControlsScrollContent.scroll_content should have contained an Entity with a ScrollContent component.");
+
+        let quiet_update_scroll_content = scroll_content.bypass_change_detection();
+
+        let (scroll_bar, scroll_bar_node) = scroll_bars_query
+            .get(scroll_handle_parent.get())
+            .expect("ScrollHandle should have had a ScrollBar parent.");
+
+        let ratio = scroll_bar_node.size() / scroll_content_node.size();
+
+        match scroll_bar.orientation {
+            ScrollBarOrientation::Vertical => {
+                quiet_update_scroll_content.current_offset.y = scroll_handle.position / ratio.y;
+            }
+            ScrollBarOrientation::Horizontal => {
+                quiet_update_scroll_content.current_offset.x = scroll_handle.position / ratio.x;
+            }
+        };
+    }
+}
```

Comment:
`scroll_handles_that_control_scroll_content` and `scroll_content_that_controls_scroll_handles` have a circular data dependency. I would like `ScrollBar + ScrollHandle` and `ScrollRoot + ScrollContent` to each be standalone widgets. If updating the state of one updates the state of the other (inputs on the `ScrollContent` node modify the `ScrollBar` node, inputs on the `ScrollBar` node modify the `ScrollContent` node), it becomes  tricky to use change detection to only update their properties when appropriate to do so.

You can see here I'm just bypassing change detection to avoid, but the whole setup seems fragile to me. I couldn't get it working such that my systems that extract widget state to `Style` could rely on change detection - they would either run more or less often than expected.